opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -2022,7 +2022,7 @@ bool CpGrid::adapt()
         is_global_refine = true; // parallel
     }
 #endif
-    if ( (comm().size() == 0) && (local_marked_elem_count == current_view_data_-> size(0)) ) {
+    if ( (comm().size() == 1) && (local_marked_elem_count == current_view_data_-> size(0)) ) {
```

Comment:
This part still looks a bit strange.

I am wondering whether we need the `#if HAVE_MPI ... #endif` above. If would hope that the code inside the #if statement would work even without MPI. In that case we can skip the if-statement and the code here.

Reply:
Good point, it seems to work without if-MPI and reducing the if-statement.